sdks/wasm/ProxyDriver/TestHardnessStartup.cs

```diff
@@ -0,0 +1,188 @@
+					throw new Exception ("node.js timedout");
+				}
+				var con_str = await tcs.Task;
+				Console.WriteLine ($"lauching proxy for {con_str}");
+
+				var proxy = new MonoProxy ();
+				var browserUri = new Uri (con_str);
+				var ideSocket = await context.WebSockets.AcceptWebSocketAsync ();
+
+				await proxy.Run (browserUri, ideSocket);
+				Console.WriteLine("Proxy done");
+			} catch (Exception e) {
+				Console.WriteLine ("got exception {0}", e.GetType ().FullName);
+			} finally {
+				proc.CancelErrorRead ();
+				proc.CancelOutputRead ();
+				proc.Kill ();
+				proc.WaitForExit ();
+				proc.Close ();
+			}
+		}
+		// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+		public void Configure (IApplicationBuilder app, IHostingEnvironment env)
+		{
+			app.UseWebSockets ();
+			app.UseStaticFiles ();
+
+			var chromePath = configuration ["ChromePath"];
+			var filesPath = configuration ["AppPath"];
+			var pagePath = configuration ["PagePath"];
+
+			Console.WriteLine ("Chrome from: '{0}'", chromePath);
+			Console.WriteLine ("Files from: '{0}'", filesPath);
+			Console.WriteLine ("Using page : '{0}'", pagePath);
+
+			var provider = new FileExtensionContentTypeProvider();
+			provider.Mappings [".wasm"] = "application/wasm";
+
+			app.UseStaticFiles (new StaticFileOptions {
+				FileProvider = new PhysicalFileProvider (filesPath),
+				ServeUnknownFileTypes = true, //Cuz .wasm is not a known file type :cry:
+				RequestPath = "",
+				ContentTypeProvider = provider
+			});
+
+
+			var psi = new ProcessStartInfo ();
+			psi.Arguments = $"--headless --disable-gpu --remote-debugging-port=9333 http://localhost:9300/{pagePath}";
+			psi.UseShellExecute = false;
+			psi.FileName = chromePath;
+			psi.RedirectStandardError = true;
+			psi.RedirectStandardOutput = true;
+
+			app.UseRouter (router => {
+				router.MapGet ("launch-chrome-and-connect", async context => {
+					Console.WriteLine ("New test request");
+					await LaunchAndServe (psi, context, str => {
+						//We wait for it as a signal that chrome finished launching
+						if (!str.StartsWith ("DevTools listening on ", StringComparison.Ordinal))
+							return null;
+
+						var client = new HttpClient ();
+						var res = client.GetStringAsync ("http://localhost:9333/json/list").Result;
+						Console.WriteLine ("res is {0}", res);
+						var obj = JArray.Parse (res);
+						var wsURl = obj? [0]? ["webSocketDebuggerUrl"]?.Value<string> ();
+						Console.WriteLine (">>> {0}", wsURl);
+
+						return wsURl;
+					});
+				});
+			});
+
+			if (configuration ["NodeApp"] != null) {
+				var nodeApp = configuration ["NodeApp"];
+				Console.WriteLine($"Doing the nodejs: {nodeApp}");
+				Console.WriteLine (Path.GetFullPath (nodeApp));
+				var nodeFullPath = Path.GetFullPath (nodeApp);
+				psi.Arguments = $"--inspect-brk=localhost:0 {nodeFullPath}";
+				psi.FileName = "node";
+
+				app.UseRouter (router => {
```

Comment:
the NodeList logit is similar to the UseDebugProxy logic in Startup.cs we should probably unify the code in a way that makes sense

Reply:
OK, I will think about that and try to change it in my next PR. :)